tempodb/encoding/vparquet3/block_traceql.go

```diff
@@ -1830,6 +1846,17 @@ func (c *traceCollector) KeepGroup(res *parquetquery.IteratorResult) bool {
 		case columnPathRootServiceName:
 			finalSpanset.RootServiceName = e.Value.String()
 			c.traceAttrs[traceql.NewIntrinsic(traceql.IntrinsicTraceRootService)] = traceql.NewStaticString(finalSpanset.RootServiceName)
+		case columnPathServicesMetaServiceName:
+			serviceName := e.Value.String()
+			finalSpanset.ServicesMeta = append(finalSpanset.ServicesMeta, traceql.ServiceMeta{ServiceName: serviceName})
+		case columnPathServicesMetaSpanCount:
+			spanCount := e.Value.Uint64()
+			finalSpanset.ServicesMeta[spanCountIdx].SpanCount = spanCount // TODO: there must be a better way
```

Comment:
`// TODO: there must be a better **way**`
Ah clever way to rebuild the entries.  The reason for the friction here is that the service metas are child rows of the trace, but they were added to the trace-level iterator (a nesting level above), so they got "flattened".   A more official way would be create a separate iterator for them, and have a new `serviceMetaCollector` recreate `traceql.ServiceMeta` structs. 

But this approach is fine imo.  The order of res.Entries should be reliable. So if you wanted to experiment: instead of incrementing the idxs you could create a new `traceql.ServiceMeta` every time columnPathServicesMetaServiceName is encountered.  It should be first of the three and represent a new row.

Reply:
In my tests I got all service names first, then all spanCounts and then the errorCounts, so while I'm let's say receiving the second span count, I need to know where to set it. After moving to a map data structure that was even more trickier (I could create a temporary slice to get a idx -> servicename mapping, but that's imho not very elegant).

I went with a serviceStatsCollector and iterator now.